src/imf.jl

```diff
@@ -0,0 +1,78 @@
+# This file is a part of AstroLib.jl. License is MIT "Expat".
+
+function _imf{T<:AbstractFloat}(mass::AbstractVector{T}, expon::AbstractVector{T},
+                                mass_range::AbstractVector{T})
+    ne_comp = length(expon)
+    if length(mass_range) != ne_comp + 1
+        println("Length of array mass_range is not one more than that of expon")
+        return zeros(mass)
+    end
+    integ = ones(T, ne_comp)
+    joint = ones(T, ne_comp)
+    norm = ones(T, ne_comp)
+    for i = 1:ne_comp
+        if expon[i] != -1
+            integ[i] = (mass_range[i+1]^(1 + expon[i]) - mass_range[i]^(1 + expon[i]))/
+                       (1 + expon[i])
+        else
+            integ[i] = log(mass_range[i+1]/mass_range[i])
+        end
+        if i != 1
+            joint[i] = joint[i-1]*(mass_range[i]^(expon[i-1] - expon[i]))
+        end
+    end
+    norm = (1/sum_kbn(integ.*joint))*joint
```

Comment:
Is there a particular reason why you used `sum_kbn` instead of the simpler `sum`?  Did you find inaccurate results with it?  Anyway, you can save an operation by using `joint` as numerator:
```julia
norm = joint ./ sum(integ .* joint)
```